django/contrib/messages/storage/cookie.py

```diff
@@ -47,6 +47,27 @@ def decode(self, s, **kwargs):
         return self.process_messages(decoded)
 
 
+class MessagePartSerializer:
+    def dumps(self, obj):
+        return [
+            json.dumps(
+                o,
+                separators=(",", ":"),
+                cls=MessageEncoder,
+            )
```

Comment:
I was thinking about moving `encode("latin-1")` here as well, but it turns out that `b",".join(obj)` is 30% slower than `",".join(obj)`, so such change would make it actually perform worse…